crates/bevy_ecs/src/world/mod.rs

```diff
@@ -706,16 +764,17 @@ impl World {
             let row = column.push_uninit();
             // SAFE: index was just allocated above
             column.set_unchecked(row, data);
+            // SAFE: index was just allocated above
+            *column.counters.get_mut().get_unchecked_mut(row) =
+                ComponentCounters::new(global_system_counter);
             std::mem::forget(value);
-            column
-                .get_flags_unchecked_mut(row)
-                .set(ComponentFlags::ADDED, true);
+            *column.get_counters_unchecked_mut(row) = ComponentCounters::new(global_system_counter);
```

Comment:
I think maybe these lines are duplicated?

Reply:
Good catch, I probably introduced it when rebasing from the ecs pr.